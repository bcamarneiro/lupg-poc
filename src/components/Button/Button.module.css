.button {
  @apply py-3 px-3 border rounded-lg;
}
