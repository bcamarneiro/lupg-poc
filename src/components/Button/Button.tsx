import { PropsWithChildren } from 'react';

import styles from './Button.module.css';
import clsx from 'clsx';
interface ButtonProps extends PropsWithChildren {
  className?: string;
  onClick: () => void;
}

const Button = ({ onClick, children, className = '' }: ButtonProps) => {
  return (
    <button
      onClick={onClick}
      className={clsx(className, styles.button, 'border-levelup-100 dark:border-levelup-800')}
    >
      {children}
    </button>
  );
};

export default Button;
