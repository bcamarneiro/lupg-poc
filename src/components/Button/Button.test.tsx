import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button component', () => {
  it('renders children correctly', () => {
    const { asFragment } = render(<Button onClick={() => {}}>Click me</Button>);
    expect(screen.getByText('Click me')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className', () => {
    render(
      <Button
        onClick={() => {}}
        className="custom-class"
      >
        Test
      </Button>,
    );
    const button = screen.getByRole('button');
    expect(button).toHaveClass('custom-class');
  });

  it('calls onClick handler when clicked', () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    const button = screen.getByRole('button');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
