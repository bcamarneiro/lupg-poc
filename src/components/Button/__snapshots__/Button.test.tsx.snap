// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders children correctly 1`] = `
<DocumentFragment>
  <button
    class="_button_2a1efe border-levelup-100 dark:border-levelup-800"
  >
    Click me
  </button>
</DocumentFragment>
`;
